### ISSUE TITLE
Expected OVER to begin window function

### ISSUE THREAD

**Issue by myacera100** [simolus3/drift#1738]
I defined several custom queries through sql statements. The problem happened when I run build to generate `model.g.dart` file; the build_runner gave error: `Expected OVER to begin window clause` on parsing the statement of aggregation filter clause as shown in the following snapshot.

<img width="670" alt="无标题" src="https://user-images.githubusercontent.com/81943962/157202098-664c3635-8822-4617-9e53-066c51287f5b.png">

I just copied the same statement to SQLite Expert's sql window and run it without error. What might be the cause for drift to interpret the `Aggregation filter clause` as `Window function clause`?

**Comment by myacera100** [comment]
As an alternative implementation, I made it through `COUNT(CASE WHEN ... THEN ... END)` and passed the parser

**Comment by simolus3** [comment]
Good catch, thanks! The parser expected a full window function invocation when seeing a `FILTER` clause which obviously isn't correct. I've fixed this problem in c9e22bf8d2c2960f3d1072d67cf069f7e23ead71.
